### ISSUE TITLE
`get` function heads in the wrong direction

### ISSUE THREAD

**Issue by justinmimbs** [Skinney/elm-dict-exploration#1]
Unless I'm going crazy, it seems the `get` function typically returns `Nothing`.

https://ellie-app.com/7Q98HyXGJa1/0

It looks like you meant to `compare targetKey key` in this line (rather than `compare key targetKey`):

https://github.com/Skinney/elm-dict-exploration/blob/1.0.0/src/Dict/LLRB.elm#L151

(Thanks for this by the way; it looks really cool!)

**Comment by justinmimbs** [comment]
As an aside: I was trying out the use of `<` and `>` instead of `compare`, and for what it's worth, this version is sometimes faster. When the keys are ints, it's faster; when the keys are strings, it's marginally faster in Chrome and faster in Safari; but when the keys are tuples, it's slower.

```elm
if targetKey < key then
    get targetKey left
else if targetKey > key then
    get targetKey right
else
    Just value
```

It seems counterintuitive, since using `<` and `>` would hit the Core Utils `cmp` function twice sometimes, rather than always once like using `compare` would (but maybe it makes up for it by not allocating Order objects). In any case, I thought I'd bring it up, since ints and strings are common keys. My apologies if this is something you've already looked into!

**Comment by robinheghan** [comment]
Fixed in version 1.0.1. Thank you for this!

I'll look into the comparison issue, nice thing to investigate. Thanks again!

**Comment by robinheghan** [comment]
@justinmimbs https://github.com/elm-lang/core/pull/918
